### ISSUE TITLE
Clean capabilities added to workflow

### ISSUE THREAD

**Issue by CoryMartin-NOAA** [NOAA-EMC/JEDI-Eval#37]
After a cycle runs and files are stored in R2D2, can we clean the cycle's run directory?

What about the diags/ directory? Can we remove files after they are plotted?

Plots probably need to stay. Could we have two options? plot_clean: false and clean: true?

**Comment by RussTreadon-NOAA** [comment]
**Orion test**

Set up `prhfox` using `feature/clean` at c19a764.   Cycle 2020121500 through 2020121600.  `*.png` and `*.stats` files generated.  `diags` and date/time stamped directories cleaned out at end of each cycle.   `gdas_hofx` job ran 24 tasks on 2 nodes instead of 8 nodes.   `feature/clean` is working as intended.

**Comment by RussTreadon-NOAA** [comment]
@CoryMartin-NOAA noted in PR #40 that unix `grep`, `cut`, `head` commands could be replaced with `source_yaml`.   This is a good suggestion.   Branch [feature/clean_refactor](https://github.com/NOAA-EMC/JEDI-Eval/tree/feature/clean_refactor) was created as a copy of develop at b142f6a to replace the unix commands with `source_yaml`.  This change will not alter results.   It simply reproduces existing functionality in a way more aligned with JEDI design philosophy.

**Comment by RussTreadon-NOAA** [comment]
While 544ce4a achieves the stated goal of replacing unix commands with `source_yaml`, it was done in a way which replicates sections of scripts `archive.sh` and `diags.sh` in the parent jobs `gdas_archive.sh` and `gdas_diags.sh`, respectively.   This is replication of scripting is not desirable.   

Discuss this with @CoryMartin-NOAA .  Two options were considered:
1. move clean up functionality added to jobs `gdas_archive.sh` and `gdas_diags.sh` into scripts `archive.sh` and `diags.sh`.
2. move replicated sections of `archive.sh` and `diags.sh` up to jobs `gdas_archive.sh` and `gdas_diags.sh`.

Option 2 is preferred by NCO.  The top level job sets the environment for the entire job - both parent and child(ren).  While hofxcs will not be implemented operationally, decide to adopt operational paradigm and implement option 2 in hofcxs workflow.

**Comment by RussTreadon-NOAA** [comment]
29c4731 moves the setting of the runtime environment from scripts `archive.sh` and `diags.sh` to jobs `gdas_archive.sh` and `gdas_diags.sh`.   The 2020121600 cycle was run on Orion.  The changes committed at 29c4731 work as intended.   Similar changes will be made to the remaining scripts and jobs for the sake of consistency.

**Comment by RussTreadon-NOAA** [comment]
87a4e27 moves the setting of the runtime environment from scripts `stage.sh`, `hofx.sh`, and `merge.sh` to jobs `gdas_stage.sh`, `gdas_hofx.sh`, and `gdas_merge.sh`, respectively.   The 2020121600 cycle was run Orion.  The changes committed at 87a4e27 worked at intended.   A short cycled experiment from 2020121500 to 2020121600 will be run on Orion to more fully test the changes.

**Comment by RussTreadon-NOAA** [comment]
Cycled test was successful.   Plots and statistic files were generated.   Data no longer needed was removed since `keepdata` was set to `no`.

Looking  at `gdas_*.sh` jobs in `src/hofx/rococto`, note that the scripting to set up the runtime environment is replicated in each job.   `gdas_hofx.sh` has a bit more set up to run hofx in parallel, but it also shares the common section of runtime environment setup scripting.   This redundancy is not desirable.

The common scripting to set up the runtime environment could be moved to a configuration file.   Each job sources this configuration file.   Changes to the run time environment can then be made in one file instead of multiple files.   This consolidation idea will be discussed with JEDI-Eval developers.

**Comment by RussTreadon-NOAA** [comment]
A new file, `src/hofx/cfg/setup`, was added to a working copy of feature/clean_refactor.   `setup` contains

```
export PYTHONPATH=$HOFX_HOMEDIR:$EMCPY_HOMEDIR

#---- other variables
alias source_yaml=$HOFX_HOMEDIR/hofx/bin/source_yaml
alias create_bundle=$HOFX_HOMEDIR/hofx/bin/create_bundle
alias detect_host=$HOFX_HOMEDIR/hofx/bin/detect_host
shopt -s expand_aliases

#---- get machine and setup runtime environment
set +eux
machine=${machine:-$(detect_host)}
source $HOFX_HOMEDIR/hofx/cfg/platform/$machine/JEDI
export R2D2_CONFIG=$HOFX_HOMEDIR/hofx/cfg/platform/$machine/r2d2_config.yaml
set -eux
```

This section was then removed from the `gdas_*.sh` jobs in `src/hofx/rocoto`.   The 2020121600 cycle was rewound and rerun.   All jobs ran to completion without error.

Given this, the above change was committed to `feature/clean_refactor` at ea31df4.

**Comment by CoryMartin-NOAA** [comment]
I think this has been resolved by #43 and will be closed
